pre_commit/staged_files_only.py

```diff
@@ -51,8 +51,8 @@ def _unstaged_changes_cleared(patch_dir: str) -> Generator[None, None, None]:
     tree = cmd_output('git', 'write-tree')[1].strip()
     retcode, diff_stdout_binary, _ = cmd_output_b(
         'git', 'diff-index', '--ignore-submodules', '--binary',
```

Comment:
I think `--binary` implies `--no-textconv` -- or that it's unavoidable (one of the two) -- I did some extensive research on this here: https://github.com/pre-commit/pre-commit/issues/776#issuecomment-1299159293

Reply:
I'm not seeing that locally, the `--binary` flag doesn't change results

```
$ git diff --binary

$ git diff --no-textconv
diff --git a/donut.jpeg b/donut.jpeg
index 3bd0743..d66a2f4 100644
Binary files a/donut.jpeg and b/donut.jpeg differ
```

It seems binary diffs are only shown if neither side of the diff is textconv'd: https://github.com/git/git/blob/2fc9e9ca3c7505bc60069f11e7ef09b1aeeee473/diff.c#L3558-L3595

Reply:
`diff-index` is different than `diff`

Reply:
Right you are, missed that, reverting this change.